### PR TITLE
[receiver/snmp] Snmp add integration tests

### PR DESCRIPTION
**Pull request by tamir-michaeli**
**Description:** <Describe what has changed.>

Adds integration tests for the SNMP Metric Receiver.

**Link to tracking Issue:** <Issue number if applicable>
#13409

**Testing:** <Describe what testing was performed and which tests were added.>
Added integration tests for scraper functionality

**Documentation:** <Describe the documentation added.>
No new documentation needed

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
pinging @djaglowski @StefanKurek as code owners

File: receiver/snmpreceiver/testdata/integration/v3_config_expected_metrics.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            "version": "latest"
+          },
+          "metrics": [
+            {
+              "description": "System uptime",
+              "name": "snmp.test.metric.sysuptime",
+              "gauge": {
+                "dataPoints": [
+                  {
+                    "aggregationTemporality": "AGGREGATION_TEMPORALITY_CUMULATIVE",
```
Comment (djaglowski) [inline receiver/snmpreceiver/testdata/integration/v3_config_expected_metrics.json:17]: This seems wrong. Gauge metrics should not have a value for aggregation temporality. I don't see anything wrong with the test.

@StefanKurek, any idea what's going on here? Might be a bug in the scraper.

Comment (StefanKurek) [reply receiver/snmpreceiver/testdata/integration/v3_config_expected_metrics.json:17]: @djaglowski it sounds like @tamir-michaeli manually created the response, so i believe this is just a problem with the response.

File: receiver/snmpreceiver/testdata/integration/docker/data/1.3.6.1.6.1.1.0.snmprec [hunk 2]
```diff
@@ -0,0 +1,9 @@
+1.3.6.1.2.1.1.3.0|65|398079840
+1.3.6.1.2.1.1.7.0|2|72
+1.3.6.1.2.1.1.8.0|65|398079874
+1.3.6.1.4.1.2021.10.1.4.1|4|Load-1
+1.3.6.1.4.1.2021.10.1.4.2|4|Load-2
+1.3.6.1.4.1.2021.10.1.4.3|4|Load-3
+1.3.6.1.4.1.2021.10.1.5.1|2|1
+1.3.6.1.4.1.2021.10.1.5.2|2|2
+1.3.6.1.4.1.2021.10.1.5.3|2|3
```
Comment (djaglowski) [inline receiver/snmpreceiver/testdata/integration/docker/data/1.3.6.1.6.1.1.0.snmprec:9]:
```suggestion
1.3.6.1.4.1.2021.10.1.5.3|2|3

```

File: receiver/snmpreceiver/testdata/integration/integration_test_v2c_config.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        description: test integration scrape 1
+        enum:
+          - sysuptime_test
+exporters:
+  nop:
+service:
+  pipelines:
+    metrics:
+      receivers: [snmp]
+      exporters: [nop]
```
Comment (djaglowski) [inline receiver/snmpreceiver/testdata/integration/integration_test_v2c_config.yaml:72]:
```suggestion
      exporters: [nop]

```

File: receiver/snmpreceiver/testdata/integration/integration_test_v3_config.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        description: test integration scrape 1
+        enum:
+          - sysuptime_test
+exporters:
+  nop:
+service:
+  pipelines:
+    metrics:
+      receivers: [snmp]
+      exporters: [nop]
```
Comment (djaglowski) [inline receiver/snmpreceiver/testdata/integration/integration_test_v3_config.yaml:78]:
```suggestion
      exporters: [nop]

```

File: receiver/snmpreceiver/testdata/integration/v2c_config_expected_metrics.json [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+               ],
+               "scope": {
+                  "name": "otelcol/snmpreceiver",
+                  "version": "latest"
+               }
+            }
+         ]
+      }
+   ]
+}
```
Comment (djaglowski) [inline receiver/snmpreceiver/testdata/integration/v2c_config_expected_metrics.json:216]:
```suggestion
}

```

File: receiver/snmpreceiver/testdata/integration/v3_config_expected_metrics.json [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+               ],
+               "scope": {
+                  "name": "otelcol/snmpreceiver",
+                  "version": "latest"
+               }
+            }
+         ]
+      }
+   ]
+}
```
Comment (djaglowski) [inline receiver/snmpreceiver/testdata/integration/v3_config_expected_metrics.json:216]:
```suggestion
}

```

File: receiver/snmpreceiver/testdata/integration/integration_test_v2c_config.yaml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        enum:
+          - in
+          - out
+      snmp.test.attribute.one.enum:
+        value: mood
+        description: Test enum with one options
+        enum:
+          - happy
+      snmp.test.1:
+        value: test
```
[the remark this answers is not in the document]
Comment (djaglowski) [reply receiver/snmpreceiver/testdata/integration/integration_test_v2c_config.yaml:62]: I may have missed the change, but can you please verify that this is resolved?

[the remark this answers is not in the document]
Comment (StefanKurek) [reply receiver/snmpreceiver/testdata/integration/integration_test_v2c_config.yaml:62]: @djaglowski looks to be resolved

**Comment by djaglowski** [comment]
@tamir-michaeli, this looks good - just need to fix the file endings and resolve conflicts

**Review by djaglowski — APPROVED** [review]

**Pull request merged**